### PR TITLE
[BEAM-11791] Continuously running microbenchmark for FnApiRunner

### PR DESCRIPTION
**Pull request by pabloem**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Dataflow/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by pabloem** [comment]
Run Seed Job

**Comment by pabloem** [comment]
Run Python Load Tests FnApiRunner Microbenchmark

**Comment by pabloem** [comment]
Run Python Load Tests FnApiRunner Microbenchmark

**Comment by pabloem** [comment]
Run Seed Job

**Comment by pabloem** [comment]
Run Seed Job

**Comment by pabloem** [comment]
Run Seed Job

**Comment by pabloem** [comment]
Run Seed Job

**Comment by pabloem** [comment]
Run Seed Job

**Comment by pabloem** [comment]
Run Python Load Tests FnApiRunner Microbenchmark

**Comment by pabloem** [comment]
Run Seed Job

**Comment by pabloem** [comment]
Run Python Load Tests FnApiRunner Microbenchmark

**Comment by pabloem** [comment]
Run Python Load Tests FnApiRunner Microbenchmark

**Comment by pabloem** [comment]
Run Seed Job

**Comment by pabloem** [comment]
New benchmark: https://ci-beam.apache.org/job/beam_Python_LoadTests_FnApiRunner_Microbenchmark_PR/4/
Passing seed job: https://ci-beam.apache.org/job/beam_SeedJob/7621/

**Comment by pabloem** [comment]
Run PythonDocker PreCommit

**Comment by pabloem** [comment]
r: @tysonjh

**Comment by tysonjh** [comment]
Is there a BEAM issue related to this, or a reason why the benchmark is being introduced & tracked over time?

File: sdks/python/apache_beam/testing/load_tests/microbenchmarks_test.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+This is load test that runs a set of basic microbenchmarks for the Python SDK
+and the DirectRunner.
```
Comment (tysonjh) [inline sdks/python/apache_beam/testing/load_tests/microbenchmarks_test.py:20]:
```suggestion
This is a load test that runs a set of basic microbenchmarks for the Python SDK
and the DirectRunner.
```

**Review by tysonjh — COMMENTED** [review]
Hard to tell if the config is correct w/o running it locally myself. Ideally you would test and confirm the changes in a local environment.

Are the instructions at https://github.com/apache/beam/blob/master/.test-infra/metrics/README.md still relevant?

**Comment by pabloem** [comment]
The reason to add this is to track performance of the Python FnApiRunner as we add support for streaming on it.
Parent issue: https://issues.apache.org/jira/browse/BEAM-7514
Subtask to add microbenchmark: https://issues.apache.org/jira/browse/BEAM-11791

**Comment by pabloem** [comment]
> Hard to tell if the config is correct w/o running it locally myself. Ideally you would test and confirm the changes in a local environment.
> 
> Are the instructions at https://github.com/apache/beam/blob/master/.test-infra/metrics/README.md still relevant?

I was unable to get influxdb to accept external connections... - I plan to use this dashboard over time. Do you think it would be okay if we bring it in, and I iterate on it to fix any issues?

**Comment by tysonjh** [comment]
> > Hard to tell if the config is correct w/o running it locally myself. Ideally you would test and confirm the changes in a local environment.
> > Are the instructions at https://github.com/apache/beam/blob/master/.test-infra/metrics/README.md still relevant?
> 
> I was unable to get influxdb to accept external connections... - I plan to use this dashboard over time. Do you think it would be okay if we bring it in, and I iterate on it to fix any issues?

Yes, that's OK. If you still have enough info around to create a bug for testing locally it would be good to document.

**Review by tysonjh — APPROVED** [review]

**Comment by tysonjh** [comment]
Also, will you squash merge or rebase? What's the plan to clean up the commit history?

**Comment by pabloem** [comment]
I've squashed all commits. I'll merge normally once tests pass

**Comment by pabloem** [comment]
Run Python PreCommit

**Pull request merged**